internal/pkg/docker/orchestrator/orchestrator.go

```diff
@@ -162,6 +162,10 @@ func (a *runTaskAction) Do(o *Orchestrator) error {
 		if err := o.stopTask(ctx, o.curTask); err != nil {
 			return fmt.Errorf("stop existing task: %w", err)
 		}
+
+		// ensure that containers are fully stopped after o.stopTask finishes blocking
+		// TODO(Aiden): Implement a container ID system or use `docker ps` to ensure containers are stopped
+		time.Sleep(1 * time.Second)
```

Comment:
ah, what error do you see otherwise - something like "xx container is already running"? Is the "xx" consistent?

Reply:
We'd see the following error:
```
[front-end] docker: Error response from daemon: Conflict. The container name "/basic-test-front-end-front-end" is already in use by container "1e49e6b3407f538ac362fd29607c9e38a1ed0309aae2a455a8ee63eea6e38b25". You have to remove (or rename) that container to be able to reuse that name.
[front-end] See 'docker run --help'.
```

Reply:
Danny said we could fix this by running `docker ps` to check when it's stopped but I think that tagging containers with a random id would be simpler and easier to implement. I can see that maybe users expect their containers to be named the same as it would be on ECS though.

Reply:
ah I see, `docker stop` returns after the container is stopped, but this doesn't rm the container. To do this, `docker rm` is probably required, which is automatically run (because we've started the container by `docker run --rm` iirc) but not waited for.

Both `docker ps -a` and random container name (maybe by appending random string) sound like good solutions that are worth further considerations!

Reply:
> with a random id would be simpler and easier to implement.

probably yes - with the downside of polluting someone's docker engine with a bunch of tags.